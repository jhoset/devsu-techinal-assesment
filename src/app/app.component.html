
<app-navbar></app-navbar>
<h1 class="text-red-400 text-3xl"> Bank </h1>
<ul>
  <li> <a routerLink="/products" > Products </a> </li>
  <li> <a routerLink="/products/add" > Add </a> </li>
  <li> <a routerLink="/products/edit" > Edit </a> </li>
</ul>
<hr>
<router-outlet></router-outlet>
