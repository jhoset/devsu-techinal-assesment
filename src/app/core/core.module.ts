import {NgModule} from '@angular/core';
import {CommonModule} from '@angular/common';
import {provideHttpClient, withInterceptors} from "@angular/common/http";
import {errorInterceptor} from "./interceptors/error.interceptor";


@NgModule({
  declarations: [],
  imports: [
    CommonModule
  ],
  providers: [
    provideHttpClient(withInterceptors([errorInterceptor]))
  ]
})
export class CoreModule {
}
