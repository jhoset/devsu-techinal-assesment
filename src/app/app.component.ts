import { Component } from '@angular/core';
import {DialogService} from "./shared/services/dialog.service";

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrl: './app.component.scss'
})
export class AppComponent {
  title = 'bank-client';
  constructor(public _dialogService: DialogService) {
  }
}
