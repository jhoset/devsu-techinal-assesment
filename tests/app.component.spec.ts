import {AppComponent} from "../src/app/app.component";
import {ComponentFixture, TestBed} from "@angular/core/testing";
import {DialogService} from "../src/app/shared/services/dialog.service";
import {Component} from "@angular/core";
import {RouterTestingModule} from "@angular/router/testing";


@Component({selector: 'app-navbar', template: 'App Dialog'})
class MockAppNav {
}

@Component({selector: 'app-dialog', template: 'App Dialog'})
class MockDialogComponent {
}

@Component({selector: 'app-toast', template: 'App Toast'})
class MockToastComponent {
}

describe('AppComponent', () => {
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;
  let dialogService: DialogService;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [AppComponent, MockDialogComponent, MockToastComponent, MockAppNav],
      providers: [DialogService],
      imports: [RouterTestingModule.withRoutes([])]
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;
    dialogService = TestBed.inject(DialogService)
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  test('should match snapshot', () => {
    fixture.detectChanges();
    expect(compiled).toMatchSnapshot();
  });

  test('should create the component', () => {
    expect(component).toBeTruthy();
  });

  test('should render app-dialog if showDialog is true', () => {
    dialogService.showDialog = true;
    fixture.detectChanges();
    expect(compiled.querySelector('app-dialog')).toBeTruthy();
  });

  test('should not render app-dialog if showDialog is false', () => {
    dialogService.showDialog = false;
    fixture.detectChanges();
    expect(compiled.querySelector('app-dialog')).toBeNull();
  });

  test('should render app-toast', () => {
    expect(compiled.querySelector('app-toast')).toBeTruthy();
  });

  test('should render app-navbar', () => {
    expect(compiled.querySelector('app-navbar')).toBeTruthy();
  });
});
