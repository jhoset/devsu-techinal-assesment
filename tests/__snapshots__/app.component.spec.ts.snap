// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `
<div
  id="root0"
>
  
  <div
    clickdetector=""
  >
    <app-navbar>
      App Dialog
    </app-navbar>
    <main
      class="h-[calc(100vh-90px)] overflow-y-auto p-4 lg:p-8"
    >
      <section
        class="max-w-[1200px] mx-auto"
      >
        <router-outlet />
        
        <app-toast>
          App Toast
        </app-toast>
      </section>
    </main>
  </div>
</div>
`;
